cmd/machine-config-daemon/pivot.go

```diff
@@ -193,25 +194,29 @@ func podmanRemove(cid string) {
 }
 
 // getDefaultDeployment uses rpm-ostree status --json to get the current deployment
-func getDefaultDeployment() types.RpmOstreeDeployment {
+func getDefaultDeployment() (types.RpmOstreeDeployment, error) {
 	// use --status for now, we can switch to D-Bus if we need more info
 	var rosState types.RpmOstreeState
+	var errRet types.RpmOstreeDeployment
 	output := utils.RunGetOut("rpm-ostree", "status", "--json")
 	if err := json.Unmarshal([]byte(output), &rosState); err != nil {
-		glog.Fatalf("Failed to parse `rpm-ostree status --json` output: %v", err)
+		return errRet, errors.Wrapf(err, "Failed to parse `rpm-ostree status --json` output")
 	}
 
 	// just make it a hard error if we somehow don't have any deployments
 	if len(rosState.Deployments) == 0 {
-		glog.Fatalf("Not currently booted in a deployment")
+		return errRet, errors.New("Not currently booted in a deployment")
 	}
 
-	return rosState.Deployments[0]
+	return rosState.Deployments[0], nil
 }
 
 // pullAndRebase potentially rebases system if not already rebased.
-func pullAndRebase(container string) (imgid string, changed bool) {
-	defaultDeployment := getDefaultDeployment()
+func pullAndRebase(container string) (imgid string, changed bool, err error) {
+	defaultDeployment, err := getDefaultDeployment()
+	if err != nil {
+		return
```

Comment:
do we want to warn since the callee isn't logging?

Reply:
The whole point of this is that the toplevel callee is catching the error right?  `pullAndRebase` :arrow_right: `run` :arrow_right: `Execute` which logs.